gym-service/src/test/java/org/example/integration/steps/trainee/TraineeSteps.java

```diff
@@ -0,0 +1,161 @@
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.example.integration.util.Constants.AUTHORIZATION_HEADER;
+import static org.example.integration.util.Constants.CONTENT_TYPE_JSON;
+
+@RequiredArgsConstructor
+public class TraineeSteps {
+    private static final String TRAINEE_PROFILE_URL = "/trainee/profile";
+    private static final String CHANGE_PASSWORD_URL = "/trainee/profile/changePassword";
+    private static final String TOGGLE_STATUS_URL = "/trainee/profile/toggleStatus";
+
+
+    private final TestContext testContext;
+    private final MockMvc mockMvc;
+    private final UserRepository userRepository;
+    private MvcResult mvcResult;
+
+    @Given("the trainee is not authenticated with username {string}")
+    public void theTraineeIsNotAuthenticated(String username) {
+        testContext.setJwtToken(null);
+    }
+
+    @When("the trainee requests the trainee profile")
+    public void theTraineeRequestsTheTraineeProfile() throws Exception {
+        mvcResult = mockMvc.perform(MockMvcRequestBuilders.get(TRAINEE_PROFILE_URL)
+                        .accept(MediaType.APPLICATION_JSON)
+                        .header(HttpHeaders.AUTHORIZATION, AUTHORIZATION_HEADER + testContext.getJwtToken())
+                        .contentType(CONTENT_TYPE_JSON))
+                .andReturn();
+    }
+
+    @When("the trainee requests to change the password with a valid new password {string}")
+    public void theTraineeRequestsToChangeThePasswordWithAValidNewPassword(String password) throws Exception {
+        performPasswordChangeRequest(password);
+    }
+
+    @When("the trainee requests to change the password with an invalid new password {string}")
+    public void theTraineeRequestsToChangeThePasswordWithAnInvalidNewPassword(String password) throws Exception {
+        performPasswordChangeRequest(password);
+    }
+
+    private void performPasswordChangeRequest(String password) throws Exception {
+        String passwordJson = new JSONObject().put("password", password).toString();
+        mvcResult = mockMvc.perform(MockMvcRequestBuilders.put(CHANGE_PASSWORD_URL)
+                        .header(HttpHeaders.AUTHORIZATION, AUTHORIZATION_HEADER + testContext.getJwtToken())
+                        .content(passwordJson)
+                        .contentType(CONTENT_TYPE_JSON))
+                .andReturn();
+    }
+
+    @When("the trainee requests to update the profile with complete data:")
+    public void theTraineeRequestsToUpdateTheProfileWithCompleteData(DataTable dataTable) throws Exception {
+        performProfileUpdateRequest(dataTable);
+    }
+
+    @When("the trainee requests to update the profile with incomplete data:")
+    public void theTraineeRequestsToUpdateTheProfileWithIncompleteData(DataTable dataTable) throws Exception {
+        performProfileUpdateRequest(dataTable);
```

Comment:
again, nothing says something is complete or incomplete here. use proper scenario namings and make advance of "given". For example, "given: trainee has incomplete update request data" and underneath this incorrect data is built by code. Last is overkill, though. Proper scenario description is enough